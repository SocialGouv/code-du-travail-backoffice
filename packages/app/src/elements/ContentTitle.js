import styled from "@emotion/styled";
import PropTypes from "prop-types";
import React from "react";

const StyledContentTitle = styled.h3`
  color: black;
  font-size: 1.1rem;
  font-weight: 600;
  margin: ${props => (props.isFirst ? "0 0 0.5rem" : "1.5rem 0 0.5rem")};
  user-select: none;
`;

const ContentTitle = ({ isFirst = false, ...props }) => (
  <StyledContentTitle isFirst={isFirst} {...props} />
);

ContentTitle.propTypes = {
  isFirst: PropTypes.bool,
};

export default ContentTitle;
