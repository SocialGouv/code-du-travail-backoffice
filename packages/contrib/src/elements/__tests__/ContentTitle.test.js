import React from "react";
import { render } from "@testing-library/react";
import ContentTitle from "../ContentTitle";

describe("[Contrib] elements/<ContentTitle />", () => {
  it("should match snapshot", () => {
    const { container } = render(<ContentTitle />);

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with `isFirst` prop", () => {
    const { container } = render(<ContentTitle isFirst />);

    expect(container).toMatchSnapshot();
  });
});
