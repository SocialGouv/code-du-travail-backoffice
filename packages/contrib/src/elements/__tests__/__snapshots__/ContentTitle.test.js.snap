// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] elements/<ContentTitle /> should match snapshot 1`] = `
<div>
  <h3
    class="ContentTitle-wfzz0s-0 hBmEVW"
  />
</div>
`;

exports[`[Contrib] elements/<ContentTitle /> should match snapshot with \`isFirst\` prop 1`] = `
<div>
  <h3
    class="ContentTitle-wfzz0s-0 fptxKe"
  />
</div>
`;
